pkg/sysregistriesv2/system_registries_v2_test.go

```diff
@@ -463,3 +463,18 @@ func TestTryUpdatingCache(t *testing.T) {
 	assert.Nil(t, registries)
 	assert.Equal(t, 1, len(configCache))
 }
+
+func TestRegistriesConfDirectory(t *testing.T) {
+	ctx := &types.SystemContext{
+		SystemRegistriesConfPath:    "testdata/base-for-registries.d.conf",
+		SystemRegistriesConfDirPath: "testdata/registries.conf.d",
+	}
+	configCache = make(map[string]*V2RegistriesConf)
```

Comment:
One more thing: `configCache` now must be keyed by _both_ the `.conf` and the `.d/*` paths, otherwise a call with different `SystemRegistriesConfDirPath` could incorrectly use previously-cached data.

Reply:
I'd expect users to invalidate the cache. We already have a very similar situation when updating `registries.conf`; we don't check time stamps or anything but force users to invalidate the cache (as cri-o does).

Reply:
Otherwise, looking up registries would get _much_ more expensive, especially with the additional `registries.conf.d.` walk.